cmd/operator-sdk/generate/internal/genutil.go

```diff
@@ -66,65 +67,82 @@ func PluginKeyToOperatorType(pluginKey string) projutil.OperatorType {
 	return ""
 }
 
-// WriteCRDs writes each CustomResourceDefinition in crds to w.
-func WriteCRDs(w io.Writer, crds ...v1beta1.CustomResourceDefinition) error {
-	for _, crd := range crds {
-		if err := writeCRD(w, crd); err != nil {
+// WriteObjects writes each CustomResourceDefinition in crds to w.
+func WriteObjects(w io.Writer, objs ...interface{}) error {
+	for _, obj := range objs {
+		if err := writeObject(w, obj); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-// WriteCRDFiles creates dir then writes each CustomResourceDefinition in crds
-// to a file in dir.
-func WriteCRDFiles(dir string, crds ...v1beta1.CustomResourceDefinition) error {
+// WriteObjectsToFiles creates dir then writes each object in crds to a file in dir.
+func WriteObjectsToFiles(dir string, objs ...interface{}) error {
 	if err := os.MkdirAll(dir, 0700); err != nil {
 		return err
 	}
-	for _, crd := range crds {
-		if err := writeCRDFile(dir, crd, makeCRDFileName(crd)); err != nil {
+	for _, obj := range objs {
+		var fileName string
+		switch t := obj.(type) {
+		case apiextv1.CustomResourceDefinition:
+			fileName = makeCRDFileName(t.Spec.Group, t.Spec.Names.Plural)
+		case apiextv1beta1.CustomResourceDefinition:
+			fileName = makeCRDFileName(t.Spec.Group, t.Spec.Names.Plural)
+		default:
+			return fmt.Errorf("unknown object type: %T", t)
+		}
+		if err := writeObjectToFile(dir, obj, fileName); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func makeCRDFileName(crd v1beta1.CustomResourceDefinition) string {
-	return fmt.Sprintf("%s_%s.yaml", crd.Spec.Group, crd.Spec.Names.Plural)
+func makeCRDFileName(group, resource string) string {
+	return fmt.Sprintf("%s_%s.yaml", group, resource)
 }
 
-// WriteCRDFilesLegacy creates dir then writes each CustomResourceDefinition
-// in crds to a file in legacy format in dir.
-func WriteCRDFilesLegacy(dir string, crds ...v1beta1.CustomResourceDefinition) error {
+// WriteObjectsToFilesLegacy creates dir then writes each object in crds to a
+// file in legacy format in dir.
+func WriteObjectsToFilesLegacy(dir string, objs ...interface{}) error {
 	if err := os.MkdirAll(dir, 0700); err != nil {
 		return err
 	}
-	for _, crd := range crds {
-		if err := writeCRDFile(dir, crd, makeCRDFileNameLegacy(crd)); err != nil {
+	for _, obj := range objs {
+		var fileName string
+		switch t := obj.(type) {
+		case apiextv1.CustomResourceDefinition:
+			fileName = makeCRDFileNameLegacy(t.Spec.Group, t.Spec.Names.Plural)
+		case apiextv1beta1.CustomResourceDefinition:
+			fileName = makeCRDFileNameLegacy(t.Spec.Group, t.Spec.Names.Plural)
+		default:
+			return fmt.Errorf("unknown object type: %T", t)
+		}
+		if err := writeObjectToFile(dir, obj, fileName); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func makeCRDFileNameLegacy(crd v1beta1.CustomResourceDefinition) string {
-	return fmt.Sprintf("%s_%s_crd.yaml", crd.Spec.Group, crd.Spec.Names.Plural)
+func makeCRDFileNameLegacy(group, resource string) string {
```

Comment:
I assume group/resource is enough to differentiate, and there's no reason to have a v1beta1 AND v1 CRD for the same resource (that probably doesn't even work). Even if that's not valid, if a user does provide them which one wins the write?

Reply:
Added a check for duplicate v1 and v1beta1 file names just to be sure.